src/Installer/dotnetup/Commands/SetInstallRoot/SetInstallRootCommand.cs

```diff
@@ -0,0 +1,219 @@
+    private bool HandleWindowsAdminPath()
+    {
+        // Check if admin PATH needs to be changed
+        if (WindowsPathHelper.AdminPathContainsProgramFilesDotnet(out var foundDotnetPaths))
+        {
+            if (foundDotnetPaths.Count == 1)
+            {
+                Console.WriteLine($"Removing {foundDotnetPaths[0]} from system PATH.");
+            }
+            else
+            {
+                Console.WriteLine("Removing the following dotnet paths from system PATH:");
+                foreach (var path in foundDotnetPaths)
+                {
+                    Console.WriteLine($"  - {path}");
+                }
+            }
+
+            if (Environment.IsPrivilegedProcess)
+            {
+                // We're already elevated, modify the admin PATH directly
+                using var pathHelper = new WindowsPathHelper();
+                pathHelper.RemoveDotnetFromAdminPath();
+            }
+            else
+            {
+                // Not elevated, shell out to elevated process
+                Console.WriteLine("Launching elevated process to modify system PATH...");
+
+                bool succeeded = WindowsPathHelper.StartElevatedProcess("elevatedadminpath removedotnet");
+                if (!succeeded)
+                {
+                    Console.Error.WriteLine("Warning: Elevation was cancelled. Admin PATH was not modified.");
+                    return false;
+                }
+            }
+        }        
+
+        return true;
+    }
+
+    private int SetUserInstallRoot()
+    {
+        // Get the default user dotnet installation path
+        string userDotnetPath = _dotnetInstaller.GetDefaultDotnetInstallPath();
+
+        Console.WriteLine($"Setting up user install root at: {userDotnetPath}");
+
+
+        // Add the user dotnet path to the user PATH
+        try
+        {
+            // On Windows, check if we need to modify the admin PATH
+            if (OperatingSystem.IsWindows())
+            {
+                if (!HandleWindowsAdminPath())
+                {
+                    //  UAC prompt was cancelled
+                    return 1;
+                }
+            }
+
+            Console.WriteLine($"Adding {userDotnetPath} to user PATH...");
+
+            var userPath = Environment.GetEnvironmentVariable("PATH", EnvironmentVariableTarget.User) ?? string.Empty;
+            var pathEntries = userPath.Split(Path.PathSeparator, StringSplitOptions.RemoveEmptyEntries).ToList();
+
+            // Check if the user dotnet path is already in the user PATH
+            bool alreadyExists = pathEntries.Any(entry =>
+                Path.TrimEndingDirectorySeparator(entry).Equals(
+                    Path.TrimEndingDirectorySeparator(userDotnetPath),
+                    StringComparison.OrdinalIgnoreCase));
+
+            if (!alreadyExists)
+            {
+                // Add to the beginning of PATH
+                pathEntries.Insert(0, userDotnetPath);
```

Comment:
Here we need to read the unexpanded user PATH value from the registry and modify that.  Use helper methods from WindowsPathHelper as possible.